src/card/collapsible/CollapsibleCard.tsx

```diff
@@ -0,0 +1,42 @@
+import "./_collapsible-card.scss";
+
+import React, {useState, useRef, useEffect} from "react";
+import classNames from "classnames";
+
+import Card from "../Card";
+
+export interface CollapsibleCardProps {
+  header: React.ReactNode;
+  isOpen: boolean;
+  children: React.ReactNode;
+  onClick: () => void;
+}
+
+function CollapsibleCard({header, isOpen, children, onClick}: CollapsibleCardProps) {
+  const [height, setHeight] = useState<number | undefined>(isOpen ? undefined : 0);
+  const ref = useRef<HTMLDivElement>(null);
```

Comment:
```suggestion
  const childrenRef = useRef<HTMLDivElement>(null);
```

Maybe it can more descriptive.